### DISCUSSION TITLE
Is it possible adonisjs orderBy on a relationship ?

### DISCUSSION THREAD

**Discussion by mdjamilaj** [adonisjs/core#3265] [Help]
the following code working on laravel...

```
$user->with(['comments' => function ($query) use ($sortDirection) {
    $query->orderBy('column', $sortDirection);
}]);
```

the following code working on Adonis 4 

```
const services = await Service
  .query()
  .with('comments')
  .orderBy('comments.id', 'desc')
  .fetch()
```

**Comment by finestgecko** [comment]
I found this discussion while searching for the same thing. Since there are no answers yet, I will leave my solution in case it's helpful for someone in the future.

You can use a sub-query in `orderBy`:

https://docs.adonisjs.com/reference/database/query-builder#using-sub-queries-4
